app/frontend/src/components/LoginButton/LoginButton.tsx

```diff
@@ -2,13 +2,23 @@ import { DefaultButton } from "@fluentui/react";
 import { useMsal } from "@azure/msal-react";
 
 import styles from "./LoginButton.module.css";
-import { getRedirectUri, loginRequest } from "../../authConfig";
-import { appServicesToken, appServicesLogout } from "../../authConfig";
+import { getRedirectUri, loginRequest, appServicesLogout, getUsername, checkLoggedIn } from "../../authConfig";
+import { useState, useEffect, useContext } from "react";
+import { LoginContext } from "../../loginContext";
 
 export const LoginButton = () => {
     const { instance } = useMsal();
+    const { loggedIn, setLoggedIn } = useContext(LoginContext);
     const activeAccount = instance.getActiveAccount();
-    const isLoggedIn = (activeAccount || appServicesToken) != null;
+    const [username, setUsername] = useState("");
+
+    useEffect(() => {
+        const fetchUsername = async () => {
+            setUsername((await getUsername(instance)) ?? "");
+        };
+
+        fetchUsername();
+    }, []);
```

Comment:
As I understand it, useEffect happens anytime the state changes. So this would trigger whenever username state changed? I guess since username is the only state, then [] is effectively the same as ["username"] for the triggers?

Reply:
calling useEffect like this only invokes it once in the constructor as far as I can tell. It has no dependencies so it only executes once. The reason for this one is that getUsername may need a fresh token and that's an async operation